### PR TITLE
Simplify diacratic removal; modify Latin & Greek preprocessors

### PR DESCRIPTION
**Pull request by martholomew**
I added a very simple function to remove diacritics that should also work with the other languages (besides just LA and GRC), but I do not know enough about them to be able to judge that.

Thanks!

### CODE REVIEW

**Comment by StefanVukovic99** [comment]
This looks good, just want to give some context on removing diacritics.
 
In some languages, diacritics may be optional, or used only in texts for children and learners, dictionaries, where disambiguation is necessary, etc. Lookup needs work on texts with or without them. Instead of trying combinations of diacritics on text without them, it is more efficient to have the diacritic-less variations in the dictionary, and remove them with a textPreprocessor if the text does have them. This may however introduce ambiguity, so maybe the term-reading system could be used, with readings containing the diacritics. 

In some cases (e.g. German umlauts) diacritics are not generally omitted, so they can be in the dictionary, and there is no need to remove them. 

In the case of Latin, I was under the impression that the long vowel marks and such were optional, but used in some contexts, and so should be removed from the dictionary and in text preprocessing. Unlike me, you seem to actually know something about Latin, so I hope you can verify these assumptions. I'll also probably update the latin diacritic removal at [KTY](https://github.com/themoeway/kaikki-to-yomitan/blob/6479532152b2a3148a63ec39081089f16816d521/4-make-yomitan.js#L396-L403) to match what you do here.

**Comment by martholomew** [comment]
I must prefix this by saying that I am a learner of Latin myself, but, as you say, the usage of macrons is not consistent in anything aside from dictionaries and learner texts.

**Comment by toasted-nutbread** [comment]
Relevant, but I've also ran into problems using string.normalize before since it is somewhat indiscriminate with regard to what it replaces. Can create problems with dictionary forms of words not being searchable.

https://github.com/themoeway/yomitan/blob/9b5de0d2d4cb224751c57bdae6558a046351c2f4/ext/js/dictionary/dictionary-importer.js#L784-L792

Realistically, there should be some guidelines for dictionary creators for how the data should be formatted or normalized.

**Comment by martholomew** [comment]
Good information for not using it too willy-nilly. I assume that it should perform well on Lain-based languages (and I did not run into any problems myself), and issues related to it would crop up quickly in languages with so few characters.

**Comment by djahandarie** [comment]
@martholomew Could you leave a similar warning on this function as a comment? Otherwise I feel it's quite likely someone might try to use it for a language with a more complicated writing system and introduce a bug by accident.

File: ext/js/language/text-preprocessors.js [hunk 1]
```diff
@@ -33,3 +33,17 @@ export const capitalizeFirstLetter = {
     options: basicTextPreprocessorOptions,
     process: (str, setting) => (setting ? str.charAt(0).toUpperCase() + str.slice(1) : str)
 };
+
+/**
+ * WARNING: This should NOT be used with languages that use Han characters,
+ *          as it can result in undesirable normalization: 
+ *            - '\u9038'.normalize('NFC') => '\u9038' (逸)
```
Comment (Scrub1492) [inline ext/js/language/text-preprocessors.js:40]: Should change to normalize('NFD'), as the code below uses NFD.

**Review by Scrub1492 — APPROVED** [review]

**Review by toasted-nutbread — DISMISSED** [review]

**Comment by toasted-nutbread** [comment]
CI unit test failure:

"ext/js/language/la/latin-text-preprocessors.js",
needs to be added to the eslintrc override with the `"serviceworker": true` environment.

**Comment by StefanVukovic99** [comment]
Maybe just needs `language-descriptors.d.ts` to be updated

**Comment by martholomew** [comment]
> @martholomew Are you able to figure out this TS error in CI or should I try to find someone who can help?

Sorry for the late response, I don't know how to fix the error, if I could get help that would be great!

**Comment by StefanVukovic99** [comment]
made a PR https://github.com/martholomew/yomitan/pull/1

**Review by jamesmaa — APPROVED** [review]

**Pull request merged**